### ISSUE TITLE
render-result-naming-convention match any function which start by render

### ISSUE THREAD

**Issue by jmfrancois** [testing-library/eslint-plugin-testing-library#537]
### Plugin version

v5.0.3

### ESLint version

v7.32.0

### Node.js version

v14.18.0

### npm/yarn version

yarn 1.22.15

### Operating system

Mac os 12

### Bug description

The following code is considered invalidate by the rule `testing-library/render-result-naming-convention`

```javascript
import {
	renderCell,
} from '../utils/gridrow';

const cellContent = renderCell(index, parent, field);
```

`cellContent` is not a recommended name for `render` returned value. Instead, you should destructure it, or name it using one of: `view`, or `utils`

### Steps to reproduce

1. create this snippet in a project with the linter setup
2. launch eslint.

### Error output/screenshots

![Screenshot 2022-01-20 at 09 17 38](https://user-images.githubusercontent.com/19857479/150299674-baecaaff-fa19-4b3b-bc17-659364d7e4a2.png)

### ESLint configuration

```{
  "extends": [
    "airbnb-typescript",
    "prettier",
    "plugin:jest-dom/recommended",
    "plugin:testing-library/react"
  ],
  "parser": "@babel/eslint-parser",
  "plugins": ["angular", "react-hooks", "@talend", "testing-library", "jest-dom"],
  "rules": {
    "@talend/import-depth": 2,
    "arrow-parens": [2, "as-needed"],
    "comma-dangle": ["error", "only-multiline"],
    "function-paren-newline": 0,
    "implicit-arrow-linebreak": 0,
    "import/prefer-default-export": 0,
    "import/no-named-as-default": "off",

    "import/no-extraneous-dependencies": [
      2,
      {
        "devDependencies": ["**/*.test.js", "**/*.test.ts*", "**/*.stories.js", "**/*.stories.ts*"],
        "peerDependencies": true
      }
    ],
    "jsx-a11y/label-has-associated-control": 2,
    "jsx-a11y/label-has-for": 0, // deprecated and buggy, replaced by jsx-a11y/label-has-associated-control
    "max-classes-per-file": 0,
    "new-cap": [2, { "capIsNewExceptions": ["List", "Map"] }],
    "no-console": [2, { "allow": ["warn", "error"] }],
    "no-else-return": [2, { "allowElseIf": true }],
    "no-mixed-operators": 0,
    "no-restricted-globals": 0,
    "no-restricted-properties": 0,
    "no-tabs": 0,
    "object-curly-newline": 0,
    "operator-linebreak": 0,
    "prefer-destructuring": 0,

    "angular/controller-name": [2, "/[A-Z].*Ctrl/"],
    "angular/di": 0,
    "angular/json-functions": 0,

    "react/button-has-type": 0,
    "react/destructuring-assignment": 0,
    "react/forbid-foreign-prop-types": 0,
    "react/forbid-prop-types": 0,
    "react-hooks/rules-of-hooks": 2,
    "react-hooks/exhaustive-deps": 1,
    "react/jsx-curly-newline": 0,
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".tsx"] }],
    "react/jsx-fragments": "off",
    "react/jsx-indent": 0,
    "react/jsx-indent-props": 0,
    "react/jsx-one-expression-per-line": 0,
    "react/jsx-props-no-spreading": "off",
    "react/jsx-wrap-multilines": 0,
    "react/no-array-index-key": 0,
    "react/no-this-in-sfc": 0, // because of ref affectation
    "react/no-unescaped-entities": 0,
    "react/no-unused-prop-types": [2, { "skipShapeProps": true }],
    "react/prefer-es6-class": 0,
    "react/require-default-props": 0,
    "react/state-in-constructor": 0,
    "react/static-property-placement": "off",

    "@typescript-eslint/indent": 0,
    "testing-library/no-node-access": "off"
  },
  "env": {
    "es6": true,
    "browser": true,
    "jasmine": true,
    "jest": true
  },
  "globals": {
    "jsdom": true,
    "TALEND_APP_INFO": true
  },
  "parserOptions": {
    "ecmaVersion": 6,
    "project": "tsconfig.json",
    "sourceType": "module"
  }
}
```

### Rule(s) affected

testing-library/render-result-naming-convention

### Anything else?

nothing

### Do you want to submit a pull request to fix this bug?

Yes

**Comment by Belco90** [comment]
Hi @jmfrancois! This false positive is related to the Aggressive Reporting feature. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior? If you are getting the error in a non-testing file, I'd suggest you restrict the plugin so [it's only run the plugin only against test files](https://github.com/testing-library/eslint-plugin-testing-library#run-the-plugin-only-against-test-files).

**Comment by jmfrancois** [comment]
Hi @Belco90 
I have read it but not understood it at the begining, but after look at the code now I understand it a bit more.
Ok after switch it off and relaunch the eslint server on vscode and the command line same output it is still consider invalid code 😬

**Comment by Belco90** [comment]
I see. If this is code from a testing file then you can only tweak the Aggressive Reporting. I'd suggest [limiting or opting out the custom renders setting](https://github.com/testing-library/eslint-plugin-testing-library#testing-librarycustom-renders). There you can find how you can limit the scope for it and a deeper explanation of how it works.

Let me know if this solves the problem.

**Comment by jmfrancois** [comment]
but adding "testing-library/custom-renders": "off" is supposed to fix the issue right ?

**Comment by Belco90** [comment]
Indeed, just setting `'testing-library/custom-renders': 'off'` should be enough. Are you adding this into the expected `settings` property within your eslintrc? Could you share your eslintrc with this option switched off perhaps?

**Comment by pstrh** [comment]
Similar issue here in conjunction with "react-test-renderer/shallow" in some older test files:

```
import { createRenderer } from "react-test-renderer/shallow";

/* The following results in:  error  `renderer` is not a recommended name for `render` returned value. 
 * Instead, you should destructure it, or  name it using  one of: `view`, 
 * or `utils`  testing-library/render-result-naming-convention 
 */
const renderer = createRenderer();
```

Fixed by:
```
// .eslintrc
{
  "settings": {
    "testing-library/custom-renders": "off"
  }
}
```
